theano/sandbox/mkl/__init__.py

```diff
@@ -0,0 +1,254 @@
+    mkl_path = os.path.abspath(os.path.split(__file__)[0])
+    mkl_ndarray_loc = os.path.join(config.compiledir, 'mkl_ndarray')
+    mkl_ndarray_so = os.path.join(
+        mkl_ndarray_loc, 'mkl_ndarray.' + get_lib_extension())
+    libmkl_ndarray_so = os.path.join(
+        mkl_ndarray_loc, 'libmkl_ndarray.' + get_lib_extension())
+
+    def try_import_mkl_ndarray():
+        mkl_files = ('mkl_ndarray.c', 'mkl_ndarray.h')
+
+        stat_times = [os.stat(os.path.join(mkl_path, mkl_file))[stat.ST_MTIME]
+                      for mkl_file in mkl_files]
+        date = max(stat_times)
+        if os.path.exists(mkl_ndarray_so):
+            if date >= os.stat(mkl_ndarray_so)[stat.ST_MTIME]:
+                return False
+
+        try:
+            sys.path[0:0] = [config.compiledir]
+            import mkl_ndarray.mkl_ndarray  # noqa
+            del sys.path[0]
+        except ImportError:
+            return False
+        return True
+
+    compile_mkl_ndarray = not try_import_mkl_ndarray()
+
+    if compile_mkl_ndarray:
+        get_lock()
+        try:
+            if not try_import_mkl_ndarray():
+                try:
+                    code = open(os.path.join(mkl_path, 'mkl_ndarray.c')).read()
+                    if not os.path.exists(mkl_ndarray_loc):
+                        os.makedirs(mkl_ndarray_loc)
+                    compiler = GCC_compiler()
+                    preargs = compiler.compile_args()
+
+                    if 'mklml_intel' in config.blas.ldflags:
+                        mkl_lib = 'mklml_intel'
+                    elif 'mklml_gnu' in config.blas.ldflags:
+                        mkl_lib = 'mklml_gnu'
+                    else:
+                        mkl_lib = 'mkl_rt'
+                    compiler.compile_str('mkl_ndarray',
+                                         code,
+                                         location=mkl_ndarray_loc,
+                                         include_dirs=[mkl_path],
+                                         libs=[mkl_lib],
+                                         preargs=preargs,)
+
+                    from mkl_ndarray.mkl_ndarray import *  # noqa
+                except Exception as e:
+                    _logger.error('Failed to compile mkl_ndarray.c: %s', str(e))
+        finally:
+            release_lock()
+
+    def ok():
+        try:
+            open(libmkl_ndarray_so).close()
+            return True
+        except IOError:
+            return False
+
+    if not ok():
+        if sys.platform == "win32":
+            shutil.copyfile(mkl_ndarray_so, libmkl_ndarray_so)
+        else:
+            try:
+                os.symlink(mkl_ndarray_so, libmkl_ndarray_so)
+            except OSError as e:
+                if getattr(e, 'errno', None) != errno.EEXIST or not ok():
+                    raise
+
+
+# register name of 'mkl_opt' in opt.py and then add tags for it.
+try:
+    from . import opt
+    opt.optdb.add_tags('mkl_opt', 'fast_compile', 'fast_run')
```

Comment:
I will remove the try and except and move the other codes within `if mkl_available()`